RecoLocalCalo/HGCalRecProducers/plugins/HGCalLayerClusterProducer.cc

```diff
@@ -63,7 +64,10 @@ HGCalLayerClusterProducer::HGCalLayerClusterProducer(const edm::ParameterSet& ps
       detector(ps.getParameter<std::string>("detector")),  // one of EE, FH, BH or "all"
       timeClname(ps.getParameter<std::string>("timeClname")),
       timeOffset(ps.getParameter<double>("timeOffset")) {
-  if (detector == "all") {
+  if (detector == "HFNose") {
+    hits_hfnose_token = consumes<HGCRecHitCollection>(ps.getParameter<edm::InputTag>("HFNoseInput"));
+    algoId = reco::CaloCluster::hfnose;
+  } else if (detector == "all") {
```

Comment:
it looks like "all" nebulously means everything but HFNose. 
Some more descriptive name would be useful.
E.g. ```HGCalAll".
Minimally, please add a comment that "all" has a special meaning